src/arrayMethodJoin.js

```diff
@@ -4,8 +4,29 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let resultStr = '';
+    let sep = separator;
+
+    if (sep === null) {
+      sep = String(sep);
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      let item = this[i];
+
+      if (item === undefined || item === null) {
+        item = '';
+      }
+
+      if (i < this.length - 1) {
+        resultStr += item + sep;
+      } else {
+        resultStr += item;
+      }
```

Comment:
Don't repeat yourself (resultStr += item): add item for all cases but add separator for all elements except the last one